sdk/messaging/azservicebus/receiver.go

```diff
@@ -358,6 +364,83 @@ func (r *Receiver) DeadLetterMessage(ctx context.Context, message *ReceivedMessa
 	return r.settler.DeadLetterMessage(ctx, message, options)
 }
 
+// DeleteMessagesOptions contains the optional parameters for the [Client.DeleteMessages] method.
+type DeleteMessagesOptions struct {
+	// Count is the maximum number of messages to delete.
+	// Defaults to 4000.
+	Count int
+
+	// BeforeEnqueueTime - any messages older than this time can be deleted.
+	// Defaults to time.Now().
+	BeforeEnqueueTime time.Time
+}
+
+// DeleteMessages deletes messages from a queue or subscription.
+// Messages are deleted on the service and are not transferred locally.
+func (r *Receiver) DeleteMessages(ctx context.Context, options *DeleteMessagesOptions) (int64, error) {
+	if options == nil {
+		options = &DeleteMessagesOptions{}
+	}
+
+	if options.BeforeEnqueueTime.IsZero() {
+		options.BeforeEnqueueTime = time.Now()
+	}
+
+	if options.Count == 0 {
+		options.Count = 4000
+	}
+
+	count := int64(0)
+
+	err := r.amqpLinks.Retry(ctx, EventReceiver, "DeleteMessages", func(ctx context.Context, lwid *internal.LinksWithID, args *utils.RetryFnArgs) error {
+		value := map[string]any{
+			"enqueued-time-utc": options.BeforeEnqueueTime.UTC(),
+			"message-count":     int32(options.Count),
```

Comment:
This makes me wonder if `Count` should be an `int32`. Will this always be limited to a 32-bit value?

Reply:
Let me check on that. I'll also adjust the Count to be an int64/int32 depending on this as well.

Reply:
So it is limited to an int32 (same value but larger type fails). I've changed Count's type to reflect that.

Reply:
The cast is now redundant.